api/src/main/java/ru/urururu/util/Iterators.java

```diff
@@ -0,0 +1,51 @@
+package ru.urururu.util;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * @author <a href="mailto:dmitriy.g.matveev@gmail.com">Dmitry Matveev</a>
+ */
+public class Iterators {
+    public static <E> Iterator<E> indexed(Function<Integer, E> getter, Supplier<Integer> lengthSupplier) {
+        return new Iterator<E>() {
+            int i = 0;
+
+            @Override
+            public boolean hasNext() {
+                return i < lengthSupplier.get();
+            }
+
+            @Override
+            public E next() {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Add a "NoSuchElementException" for iteration beyond the end of the collection. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2272)